core/src/execution/datafusion/expressions/cast.rs

```diff
@@ -100,15 +118,22 @@ impl Cast {
             .iter()
             .map(|value| match value {
                 Some(value) => match value.to_ascii_lowercase().trim() {
-                    "t" | "true" | "y" | "yes" | "1" => Some(true),
-                    "f" | "false" | "n" | "no" | "0" => Some(false),
-                    _ => None,
+                    "t" | "true" | "y" | "yes" | "1" => Ok(Some(true)),
+                    "f" | "false" | "n" | "no" | "0" => Ok(Some(false)),
+                    other if ansi_mode => {
+                        return Err(CometError::CastInvalidValue {
+                            value: other.to_string(),
+                            from_type: "STRING".to_string(),
+                            to_type: "BOOLEAN".to_string(),
+                        })
+                    }
```

Comment:
This is the actual fix to add ANSI support in this specific cast operation. The rest of the PR is plumbing to pass ANSI mode down into the planner and improving the cast test suite.